### PR TITLE
Make PrincipalListField extensible via plugins

### PR DESCRIPTION
**Pull request by micsucmed**
This PR adds the necessary signals to allow the implementation of csv import/export for peer review judges and reviewers implemented in https://github.com/indico/indico-plugin-jacow/pull/12.

### CODE REVIEW

File: indico/web/client/js/react/components/principals/PrincipalListField.jsx [hunk 1]
```diff
@@ -164,6 +166,7 @@ const PrincipalListField = props => {
               )}
             </>
           )}
+          {renderPluginComponents('setManagersButton', {fieldId, eventId})}
```
Comment (ThiefMaster) [inline indico/web/client/js/react/components/principals/PrincipalListField.jsx:169]: - Most of those signal names use kebab-case capitalization (even though it's pretty inconsistent atm :/)
- The name is way too generic, something like `principal-list-field-add-buttons` would be more suitable

File: indico/web/client/js/react/components/principals/PrincipalListField.jsx [hunk 2]
```diff
@@ -44,6 +45,7 @@ const PrincipalListField = props => {
     eventId,
     favoriteUsersController,
     className,
+    fieldId,
```
Comment (ThiefMaster) [inline indico/web/client/js/react/components/principals/PrincipalListField.jsx:48]: Can we somehow avoid that? When using this outside a WTForm (typically using `FinalPrincipalList`), there is no hidden field with an ID that could be used by the code connecting to your signal...

File: indico/modules/events/papers/forms.py [hunk 3]
```diff
@@ -39,8 +39,10 @@ class PaperTeamsForm(IndicoForm):
                                   event=lambda form: form.event,
                                   description=_('List of users allowed to manage the call for papers'))
     judges = PrincipalListField(_('Judges'),
+                                event=lambda form: form.event,
```
Comment (ThiefMaster) [inline indico/modules/events/papers/forms.py:42]: Why is this needed?

File: indico/web/client/js/react/components/principals/PrincipalListField.jsx [hunk 4]
```diff
@@ -81,6 +83,10 @@ const PrincipalListField = props => {
     onChange([...value, ...data.map(x => x.identifier)]);
     markTouched();
   };
+  const handleAddCSVEmails = data => {
```
Comment (ThiefMaster) [inline indico/web/client/js/react/components/principals/PrincipalListField.jsx:86]: that name is too specific for plugin entry point code

File: indico/web/client/js/react/components/principals/PrincipalListField.jsx (hunk 2 above)
Comment (micsucmed) [reply indico/web/client/js/react/components/principals/PrincipalListField.jsx]: I passed the `fieldId` for a `boolean` set on the `PrincipalListField` so that we can avoid this, so I had to make some changes on how to generate the CSV export file; now created on the front-end instead of the back-end

Comment (ThiefMaster) [reply indico/web/client/js/react/components/principals/PrincipalListField.jsx]: I'm not sure if that's better. The whole idea of plugin interfaces is that they are generic. `importExport` is rather specific to what this particular plugin does.

I also don't have a better idea right now though (but TBH I prefer passing the ID over this importExport boolean), but I'll see if I can come up w/ something...

Comment (tomasr8) [reply indico/web/client/js/react/components/principals/PrincipalListField.jsx]: IIUC there are two things we need to pass (for this specific feature):
 - whether the button is for import or export (`importExport`)
 - if it is an export button, whether it is for judges or reviewers (we get this information from the `fieldId`)

The `fieldId` is needed only for the second one, but we discussed with Michel and we can probably remove it. This still leaves us with the the `importExport` prop though

Comment (micsucmed) [reply indico/web/client/js/react/components/principals/PrincipalListField.jsx]: I passed the fieldId before as I could use it for both, determine if the buttons should be rendered (only for the `fieldId`s of `judges` and `content_reviewers`) and to ask for the corresponding list of users for both fields when exporting.

By generating the CSV on the front end I don't need the fieldId to determine which users to include in the exported file (since they are state variables within each PrincipalListField` component  individually), so instead I added the `importExport` prop that just tells the `PrincipalListField` if it should render the buttons for `import` AND `export` or not.

Without any flag like the `importExport` boolean getting the buttons to rendered only on the desired `PrincipalListFiled`s is not possible, and maybe the `fieldId` might seem to specific for the current fields targeted.

Comment (micsucmed) [reply indico/web/client/js/react/components/principals/PrincipalListField.jsx]: @ThiefMaster any news on this, should I just rollback to the `fieldId` solution?

Comment (ThiefMaster) [reply indico/web/client/js/react/components/principals/PrincipalListField.jsx]: I'll have a look once I'm done w/ the event/category import/export stuff... still not clear what's the best solution there, but I think when checking out the branch and trying things out it'll become clearer than from just looking at the diffs here

Comment (ThiefMaster) [reply indico/web/client/js/react/components/principals/PrincipalListField.jsx]: So I still don't like this at all, because those field ids are not even guaranteed to be unique across pages (only within a single page). I got an idea how we could do it much more nicely using react context. Will add a commit to both PRs in a bit if it works as expected.

File: indico/modules/events/papers/forms.py (hunk 3 above)
Comment (ThiefMaster) [reply indico/modules/events/papers/forms.py:43]: Nevermind, without it the PrincipalListField does not have the event id, which is needed in the plugin code in order to build the API URL.
